src/utils/url_params.js

```diff
@@ -1,19 +1,54 @@
 /**
- * parseQueryArgs - parses the query args from a given url
- * @url {string} - string of the url to parse the query args from
- * @separatorString {string} - string to separate the query args by
- * @paramChar {string} - string to split the url from the query args
- * @decodeArgs {boolean} - boolean to determine decoding the query args
+ * Parses the query args from a given url. Both #..&.. hash args as well as ?..&.. query args are supported.
+ * @param url {string} - string of the url to parse the query args from
+ * @param separatorString {string} - string to separate the query args by
+ * @param paramChar {string} - string to split the url from the query args
+ * @param decodeArgs {boolean} - if true, decodes the query arg string first (useful for encoded hash args)
  * @returns {object} - object of the query args
  */
 export function parseQueryArgs(url, separatorString = "&", paramChar = "?", decodeArgs = false) {
-    const hashAt = url.indexOf(paramChar);
-    let queryArgs = hashAt >= 0 && url.substr(hashAt+1);
+    const argsStart = url.indexOf(paramChar);
+    let queryArgs = argsStart >= 0 && url.substring(argsStart+1);
     if (decodeArgs) {
         queryArgs = decodeURIComponent(queryArgs);
     }
     return parseArgs(queryArgs, separatorString);
-};
+}
+
+/**
+ * Includes the specified parameters in the url's query args, replacing existing args as needed, adding them if missing.
+ * Both #..&.. hash args as well as ?..&.. query args are supported.
+ * @param url - the url to update
+ * @param params {Object} - key/value parameter map to include * @param separatorString
+ * @param url {string} - string of the url to parse the query args from
+ * @param separatorString {string} - string to separate the query args by
+ * @param paramChar {string} - string to split the url from the query args
+ * @returns {string}
+ */
+export function updateQueryArgs(url, params, separatorString = "&", paramChar = "?") {
+    const existingArgs = parseQueryArgs(url, separatorString, paramChar);
+    if (!params) params = {};
+    const updatedArgs = {...existingArgs, ...params};
+    return setQueryArgs(url, updatedArgs, separatorString, paramChar);
+}
+
+/**
+ * Sets the url's query args to the specified values. Both #..&.. hash args as well as ?..&.. query args are supported.
+ * @param url - the url to update
+ * @param params {Object} - key/value parameter map to include * @param separatorString
```

Comment:
merged two lines ( "@param param {Object}" and "* @param separatorString )

Reply:
good catch